Controller/PostController.php

```diff
@@ -30,16 +31,17 @@ public function homeAction()
     }
 
     /**
-     * @param array $criteria
-     * @param array $parameters
+     * @param array   $criteria
+     * @param array   $parameters
+     * @param Request $request
      *
      * @return \Symfony\Component\HttpFoundation\Response
      */
-    public function renderArchive(array $criteria = array(), array $parameters = array())
+    public function renderArchive(array $criteria = array(), array $parameters = array(), Request $request)
```

Comment:
You can maybe call `resolveReques` as done here: https://github.com/sonata-project/SonataNewsBundle/pull/257/files#diff-470b4c7629c3b1960c1c3978c9069fd9R72